spi/src/main/java/org/opensearch/ml/common/spi/tools/Tool.java

```diff
@@ -127,5 +129,7 @@ interface Factory<T extends Tool> {
          * @return the default tool version
          */
         String getDefaultVersion();
+
+        List<String> getAllModelKeys();
```

Comment:
Similarly, you can declare `List<String> getAllConnectorKeys();` for ConnectorTool.